src/collision/CollisionHandler.cpp

```diff
@@ -72,37 +75,102 @@ Entity *CollisionHandler::detectPickUpCollision(std::vector<Entity*> returnObjec
 
 }
 
-// Created by DericM 3/8/2017
-// TODO fix collision detection using new quadtrees
-std::priority_queue<const HitBox*> CollisionHandler::detectLineCollision(
-        Marine &marine, const int range) {
+
+/**
+    detectLineCollision
+
+    DISCRIPTION:
+        This checks each relevent quadTrees for targets that are in the weapons sights,
+        and adds them to a priority queue that is sorted by distance from the player.
+
+    AUTHOR: Deric Mccadden 3/8/2017
+        Edited: 3/16/2017 fixed to use new quad trees (cant shoot walls yet)
+        Edited: 3/17/2017 walls work now.
+
+    PARAMS:
+        std::priority_queue<Target>& targetsInSights,
+            This is the queue that will hold the targets that will be determined.
+        Marine &marine,
+            The marine that fired the weapon, this is needed for its x and y and angle.
+        const int range
+            The range of the weapon that is being fired.
+*/
+void CollisionHandler::detectLineCollision(
+        std::priority_queue<Target>& targetsInSights, Marine &marine, const int range) {
 
     const double degrees = marine.getAngle() - 90;
     const double radians = degrees * M_PI / 180;
-    const int playerX = marine.getX() + (MARINE_WIDTH / 2);
-    const int playerY = marine.getY() + (MARINE_HEIGHT / 2);
+    const int originX = marine.getX() + (MARINE_WIDTH / 2);
+    const int originY = marine.getY() + (MARINE_HEIGHT / 2);
     const int deltaX  = range * cos(radians);
     const int deltaY  = range * sin(radians);
-    int aX, aY, bX, bY;
 
-    std::vector<Entity*> allEntities;
+    auto& zombies = quadtreeZombie.objects;
+    auto& turrets = quadtreeTurret.objects;
+    auto& walls   = quadtreeWall.objects;
```

Comment:
These temp variables are never reused
Remove them and place their definitions inline below